apps/artemis-web/HOC/withAuth.tsx

```diff
@@ -0,0 +1,59 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { extractUser } from '../lib/helpers';
+import passport from '../lib/passport';
+import auth from '../middleware/auth';
+import { initializeApollo } from '../utils/graphql';
+import nc from 'next-connect';
+import React from 'react';
+import { useCookie } from 'next-cookie';
+import jwt from 'jsonwebtoken';
+
+const withAuth = (WrappedComponent) => {
+  const Wrapper = (props) => {
+    return <WrappedComponent {...props} />;
+  };
+
+  class App extends React.PureComponent {
+    render() {
+      return <WrappedComponent {...this.props} />;
+    }
+  }
+
+  return App;
+};
+
+export async function getProps(ctx) {
+  const apolloClient = initializeApollo(
```

Comment:
Apollo should not be together with auth. 

I guess after as it will need the access_token

Reply:
True.

Reply:
mostly because not all pages need apollo ;)